src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            _configuration = configuration;
+            _httpClient = new HttpClient();
+            _httpClient.BaseAddress = new Uri("https://support-bay-api.azurewebsites.net/observer/");
+            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+        public Task<JObject> GetAdminSitesByHostName(string stampName, string[] hostNames)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<JObject> GetAdminSitesBySiteName(string stampName, string siteName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<object>> GetAppServiceEnvironmentDeployments(string hostingEnvironmentName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<JObject> GetAppServiceEnvironmentDetails(string hostingEnvironmentName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<Dictionary<string, string>>> GetCertificatesInResourceGroup(string subscriptionName, string resourceGroupName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string siteName)
+        {
+            var response = await GetObserverResource($"sites/{siteName}/runtimesiteslotmap");
+            var slotTimeRangeCaseSensitiveDictionary = JsonConvert.DeserializeObject<Dictionary<string, List<RuntimeSitenameTimeRange>>>(response);
+            var slotTimeRange = new Dictionary<string, List<RuntimeSitenameTimeRange>>(slotTimeRangeCaseSensitiveDictionary, StringComparer.CurrentCultureIgnoreCase);
+            return slotTimeRange;
+        }
+
+        public async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string stampName, string siteName)
+        {
+            var response = await GetObserverResource($"stamp/{stampName}/sites/{siteName}/runtimesiteslotmap");
+            var slotTimeRangeCaseSensitiveDictionary = JsonConvert.DeserializeObject<Dictionary<string, List<RuntimeSitenameTimeRange>>>(response);
+            var slotTimeRange = new Dictionary<string, List<RuntimeSitenameTimeRange>>(slotTimeRangeCaseSensitiveDictionary, StringComparer.CurrentCultureIgnoreCase);
+            return slotTimeRange;
+        }
+
+        public Task<IEnumerable<Dictionary<string, string>>> GetServerFarmsInResourceGroup(string subscriptionName, string resourceGroupName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<string> GetServerFarmWebspaceName(string subscriptionId, string serverFarm)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<IEnumerable<string>> GetSiteHostNames(string siteName)
```

Comment:
Yeah given the bug I just mentioned, it seems like it is incorrect to ship this API as is, since it can give incorrect data

Reply:
Honestly, I may remove this entirely seeing that we require hostnames to be passed through the API

Reply:
Will hostnames ask be fulfilled by geomaster work?

Reply:
Even without the site information in the request body, we have them from the query parameters. Probably don't need a method to get them again, at least I can't think of a reason.